collector/memory_bsd.go

```diff
@@ -45,6 +45,13 @@ func NewMemoryCollector() (Collector, error) {
 	}
 	size := float64(tmp32)
 
+	mibSwapTotal := "vm.swap_total"
+	/* swap_total is FreeBSD specific. Fall back to Dfly specific mib if not present. */
+	_, err = unix.SysctlUint32(mibSwapTotal)
+	if err != nil {
```

Comment:
What if `vm.swap_size` is not correct either? what about other BSDs?

would it make sense to check for the specific kind of error that switching to `_size` can fix (not found, I suppose?)

Reply:
I suppose this collector only supports Dragonfly and FreeBSD. They are relatively similar to each other comparing to other BSD variants, and in this case, that mib is the only difference. If there is `_total`, implicitly it is supposed to be FreeBSD otherwise Dragonfly.